Broken on node versions 14.8.0 to 14.17.6

This pattern:

https://github.com/emberjs/data/blob/637a7d4ae202273bb4c5cfecdd5a6126a43543fa/packages/private-build-infra/src/utilities/require-module.js#L1

started working in Node 14.18.0. But the engines declaration:

https://github.com/emberjs/data/blob/637a7d4ae202273bb4c5cfecdd5a6126a43543fa/packages/private-build-infra/package.json#L48

says that all node versions down to 14.8.0 are supported. This can result in a runtime failure.

Reproduction:
 - use node 14.17.6 (I used `volta pin node@14.17.6`)
 - use ember-data 4.6.1
 - `ember b`

Observed failure:

```
$ ember build
Cannot find module 'node:fs'
Require stack:
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/node_modules/@ember-data/private-build-infra/src/utilities/require-module.js
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/node_modules/@ember-data/private-build-infra/src/packages.js
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/node_modules/@ember-data/private-build-infra/src/debug-macros.js
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/node_modules/@ember-data/private-build-infra/src/stripped-build-plugins.js
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/node_modules/@ember-data/private-build-infra/src/addon-build-config-for-data-package.js
- /Users/edward/hacking/embroider/node_modules/ember-data-latest/index.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/models/package-info-cache/package-info.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/models/package-info-cache/index.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/models/project.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/utilities/get-config.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/utilities/instrumentation.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/lib/cli/index.js
- /Users/edward/hacking/embroider/node_modules/ember-cli-latest/bin/ember
```

Apps can workaround by upgrading to Node 14.18.0 or newer.

--- a/packages/private-build-infra/src/utilities/require-module.js
+++ b/packages/private-build-infra/src/utilities/require-module.js
@@ -1,4 +1,4 @@
-const fs = require('node:fs');
+const fs = require('fs');
 
 module.exports = function requireModule(modulePath) {
   const path = require.resolve(modulePath);

